app/models/post.rb

```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :comments, foreign_key: 'post_id'
+  has_many :likes, foreign_key: 'post_id'
+
+  def update_user_posts_counter
+    author.update(post_counter: author.posts.count)
+  end
+
```

Comment:
- [OPTIONAL] Good job so far, I will suggest using one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_user_posts_counter` method every time a new post has been saved to the database, then make the `update_user_post_counter` method private 🙃. Also, you can do the same for `update_likes_counter` and `update_comments_counter`.

Reply:
Thank you for your suggestion @cynthiainga , I'll make sure to add that to my code 👍😁